src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -210,6 +199,10 @@ public string HubName
         // Used for mocking the lifecycle notification helper.
         internal HttpMessageHandler NotificationHandler { get; set; }
 
+        // This is just a way for tests to overwrite the webhook url, since there is no easy way
+        // to mock the value from ExtensionConfigContext. It should not be used in production code paths.
+        internal Uri TestWebhookUri { get; set; }
```

Comment:
It feels weird to have something named "Test" in non-test code.  I wonder if we can fix the name and make this a bit more extensible by doing something like this:

 ```suggestion
        public Func<Uri> WebhookUriProviderOverride { get; set; }
```

Then, in DurableTaskExtension we could do something like this:

```csharp
this.HttpApiHandler.RegisterWebhookProvider(
    this.Options.WebhookUriProviderOverride ?? context.GetWebhookHandler);
```